### ISSUE TITLE
Support for Google Oauth2 post G+

### ISSUE THREAD

**Issue by jeffoneill** [authomatic/authomatic#197]
Below is an attempt to update `providers/oauth2.py` so that it will work with Google login after the shutdown of G+.  It works currently with the email and profile scopes.

I'm not sure if it will continue to work post G+ shutdown.  🤞 

```
-    user_info_url = 'https://www.googleapis.com/plus/v1/people/me'
+    user_info_url = 'https://www.googleapis.com/oauth2/v1/userinfo'

     @staticmethod
     def _x_user_parser(user, data):
-        emails = data.get('emails', [])
-        if emails:
-            user.email = emails[0].get('value')
-            for email in emails:
-                if email.get('type') == 'account':
-                    user.email = email.get('value')
-                    break
-
-        user.id = data.get('sub') or data.get('id')
-        user.name = data.get('displayName')
-        user.first_name = data.get('name',{}).get('givenName')
-        user.last_name = data.get('name',{}).get('familyName')
-        user.locale = data.get('language')
-        user.link = data.get('url')
-        user.picture = data.get('image',{}).get('url')
+        user.id = data.get('id')
+        user.email = data.get('email')
+        user.name = data.get('name')
+        user.first_name = data.get('given_name')
+        user.last_name = data.get('family_name')
+        user.locale = data.get('locale')
+        user.picture = data.get('picture')
         try:
             user.birth_date = datetime.datetime.strptime(data.get('birthdate'), "%Y-%m-%d")
-        except:
+        except Exception:
             user.birth_date = data.get('birthdate')
         return user

```

**Comment by ericof** [comment]
@jeffoneill It is already working on master. Just take a look at it

**Comment by ergo** [comment]
I think google left OAuth endpoint support for now - everything is still working for us on stable authomatic.

**Comment by jeffoneill** [comment]
It may be working now, but it could break at any time if Google decides to shutdown the G+ API.

**Comment by ergo** [comment]
```
Some APIs labelled "Google+" provide access to basic account information that is critical to sign-in use cases, including for many third-party apps and sites. To help mitigate the impact of the Google+ APIs shutdown on the sign-in use cases, we have created a new implementation of several people.get and people.getOpenIdConnect APIs that will only return basic fields necessary for sign-in functionality such as name and email address, if authorized by the user. The new implementation only allows an app to retrieve the profile of the signed-in user, and can return only basic profile fields necessary for user sign-in functionality.
While we still recommend that developers migrate to alternative APIs such as Google Sign-in and Google People API, for cases where developers are unable to move over before March 7, existing calls made to the legacy Google+ people.get and people.getOpenIdConnect APIs will automatically be served by this new implementation.
```
This is from their communication, it will be there for time being. But yes - we should migrate off eventually.

**Comment by ericof** [comment]
On master we do not use Google+ api's anymore

**Comment by ericof** [comment]
Here: https://github.com/authomatic/authomatic/blob/master/authomatic/providers/oauth2.py#L1253

**Comment by mrichar1** [comment]
After several years of inactivity, `authomatic` is now under community management, and we have just released a new stable version (`1.0.0`).

We are now reviewing all issues and PRs and hoping to begin work to solve as many of these as possible.

We are keen to find out which issues still apply, and which PRs are still required/are likely to merge cleanly into the current code. We are aiming to review them all, but any help with prioritisation would be very useful!

If you are still interested in having this issue/PR resolved, or are able to help us work on it, please reply to this message. That way we know which issues are most important to the community.

**Comment by ericof** [comment]
With the new release, this is solved. Closing this issue.
